bin/drop_config.py

```diff
@@ -1,191 +1,200 @@
-    yaml_object["geneAnnotation"].pop("gtf", None)
-    yaml_object["exportCounts"]["geneAnnotations"] = [gtf_without_ext]
-    yaml_object["genomeAssembly"] = genome_assembly
-
-    ## Export counts
+from typing import Set, Dict
+
+SCRIPT_VERSION = "v1.0"
+CONFIG_YAML = {
+    "projectTitle": "DROP: Detection of RNA Outliers Pipeline",
+    "root": None,
+    "htmlOutputPath": None,
+    "indexWithFolderName": True,
+    "hpoFile": None,
+    "sampleAnnotation": None,
+    "geneAnnotation": {"gtf": None},
+    "genomeAssembly": "genome",
+    "exportCounts": {"geneAnnotations": ["gtf"], "excludeGroups": [None]},
+    "genome": None,
+    "aberrantExpression": {
+        "run": False,
+        "groups": ["outrider"],
+        "fpkmCutoff": 1,
+        "implementation": "autoencoder",
+        "padjCutoff": 0.05,
+        "zScoreCutoff": 0,
+        "genesToTest": None,
+        "maxTestedDimensionProportion": 3,
+        "yieldSize": 2000000,
+    },
+    "aberrantSplicing": {
+        "run": False,
+        "groups": ["fraser"],
+        "recount": False,
+        "longRead": False,
+        "keepNonStandardChrs": True,
+        "filter": True,
+        "minExpressionInOneSample": 20,
+        "quantileMinExpression": 10,
+        "minDeltaPsi": 0.05,
+        "implementation": "PCA",
+        "padjCutoff": 0.1,
+        "maxTestedDimensionProportion": 6,
+        "genesToTest": None,
+        "FRASER_version": "FRASER2",
+        "deltaPsiCutoff": 0.1,
+        "quantileForFiltering": 0.75,
+    },
+    "mae": {
+        "run": False,
+        "groups": ["mae"],
+        "gatkIgnoreHeaderCheck": True,
+        "padjCutoff": 0.05,
+        "allelicRatioCutoff": 0.8,
+        "addAF": True,
+        "maxAF": 0.001,
+        "maxVarFreqCohort": 0.05,
+        "qcVcf": None,
+        "qcGroups": "mae",
+        "dnaRnaMatchCutoff": 0.85,
+    },
+    "rnaVariantCalling": {
+        "run": False,
+        "groups": ["batch_0"],
+        "highQualityVCFs": [
+            "Data/Mills_and_1000G_gold_standard.indels.hg19.sites.chrPrefix.vcf.gz",
+            "Data/1000G_phase1.snps.high_confidence.hg19.sites.chrPrefix.vcf.gz",
+        ],
+        "dbSNP": "Data/00-All.vcf.gz",
+        "repeat_mask": "Data/hg19_repeatMasker_sorted.chrPrefix.bed",
+        "createSingleVCF": True,
+        "addAF": True,
+        "maxAF": 0.001,
+        "maxVarFreqCohort": 0.05,
+        "hcArgs": "",
+        "minAlt": 3,
+        "yieldSize": 100000,
+    },
+    "tools": {"gatkCmd": "gatk", "bcftoolsCmd": "bcftools", "samtoolsCmd": "samtools"},
+}
+
+
+def update_config(
+    genome: Path,
+    gtf: Path,
+    genome_assembly: str,
+    padjcutoff: float,
+    zscorecutoff: float,
+    drop_module: str,
+) -> Dict:
+    """
+    Updates config file to add correct genome, gtf,
+    adjusted p-value, and Z-score for the module to be run.
+    """
+    gtf_without_ext = gtf.stem
+    genome_name = genome.name
+
+    CONFIG_YAML["genome"] = genome_name
```

Comment:
Sorry I should have been more explicit. Constants are inmutable objects so you should not try to change them and used a copy. In this case it may be also a good idea to deep copy it, which ensures that a completely independent copy of the data structure is created, including copies of all nested objects
```suggestion
    config_copy: Dict[str, Any] = copy.deepcopy(CONFIG_YAML)
    CONFIG_YAML["genome"] = genome_name
```

Reply:
Ahhh, I thought about it but left it as was, I have now changed it :)